components/teams-service/storage/postgres/postgres.go

```diff
@@ -365,48 +353,23 @@ func (p *postgres) PurgeUserMembership(ctx context.Context, userID string) ([]uu
 func (p *postgres) AddUsers(ctx context.Context,
 	teamID uuid.UUID,
 	userIDs []string) (storage.Team, error) {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	tx, err := p.db.BeginTx(ctx, nil)
-	if err != nil {
-		return storage.Team{}, p.processError(err)
-	}
-
-	doneSomething := false
-	for _, userID := range userIDs {
-		res, err := tx.ExecContext(ctx,
-			`INSERT INTO teams_users_associations (team_id, user_id, created_at)
-			VALUES ($1, $2, now())`, teamID, userID)
-		if err != nil {
-			return storage.Team{}, p.processError(err)
-		}
-		num, err := res.RowsAffected()
-		if err != nil {
-			return storage.Team{}, p.processError(err)
-		}
-		if num > 0 {
-			doneSomething = true
-		}
-	}
-
-	var team storage.Team
-
-	if !doneSomething {
-		team, err = p.getTeam(ctx, tx, teamID)
-
-	} else {
-		team, err = p.touchTeam(ctx, tx, teamID)
-	}
-	if err != nil {
-		return storage.Team{}, p.processError(err)
-	}
-
-	err = tx.Commit()
+	var t storage.Team
+	err := p.db.QueryRowContext(ctx,
+		`WITH moved_rows AS (
+			INSERT INTO teams_users_associations (team_db_id, user_id, created_at)
+				SELECT db_id, unnest($2::TEXT[]), now()
+				FROM teams
+				WHERE id=$1
+			RETURNING team_db_id
+		)
+		UPDATE teams SET updated_at=NOW()
+		WHERE db_id in (SELECT * FROM moved_rows)
```

Comment:
There's days of potential SQL enhancements in this one file alone. Since this works, I'd keep it as-is and go on. It's not like the work done too often was much work, or heavyweight.

Reply:
I've added a `DISTINCT`